docs(core): add howto section (#2505)
SwissDataScienceCenter/renku-python: docs/how-to-guides/hpc.rst

.. _hpc:

Running Renku on HPC
====================

Renku CLI supports various backends for executing workflows. Currently, there
are two different providers are implemented, namely ``cwltool`` and ``toil``.
:ref:`provider` documents gives a more detailed description of how to implement
your own workflow provider.

> The default all workflows are executed by the ``cwltool`` provider, that basically

Reviewer: ```suggestion
By default all workflows are executed by the ``cwltool`` provider, that
```

exports the workflow to CWL and then uses `cwltool <https://github.com/common-workflow-language/cwltool>`_
to execute the given CWL.

> The workflow backend can be changed by using the ``-p/--provider <PROVIDER>`` and

Reviewer: ```suggestion
The workflow backend can be changed by using the ``-p/--provider <PROVIDER>`` 
```

command line option. A backend's default configuration can be overridden by

> providing the  ``-c/--config <config.yaml>`` a command line parameter.

Reviewer: ```suggestion
providing the  ``-c/--config <config.yaml>`` command line parameter.
```

> The following ``renku`` commands support the above mentioned, workflow provider

Reviewer: ```suggestion
The following ``renku`` commands support the above mentioned workflow provider
```

related command line options:

 - :ref:`cli-rerun`,
 - :ref:`cli-update`,

>  - :ref:`cli-workflow` execute and iterate.

Reviewer: ```suggestion
 - :ref:`cli-workflow` ``execute`` and ``iterate``.
```

For example, to execute a previously created ``my_plan`` workflow with ``toil``, one
simply would run the following command:

.. code-block:: console

   $ renku execute -p toil my_plan

Using ``toil`` as a workflow provider has the advantage that it supports running
the workflows on various `high-performance computing <https://toil.readthedocs.io/en/latest/running/hpcEnvironments.html>`_
and `cloud <https://toil.readthedocs.io/en/latest/running/cloud/cloud.html#cloud-platforms>`_
platforms.

Renku on Slurm
^^^^^^^^^^^^^^
`Slurm <https://www.schedmd.com/>`_ is a highly configurable open-source workload manager,
which is in widespread use at government laboratories, universities and companies world
wide and performs workload management for over half of the top 10 systems in the TOP500.

As ``toil`` supports Slurm, one can easily execute the previously created renku

> workflows on Slurm, just need to provide a simple configuration file to the provider

Reviewer: ```suggestion
workflows on Slurm. One just needs to provide a simple configuration file to the provider
```
